### ISSUE TITLE
Setting verbose=False issues with threads

### ISSUE THREAD

**Issue by jeberger** [abetlen/llama-cpp-python#1874]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [x] I carefully followed the [README.md](https://github.com/abetlen/llama-cpp-python/blob/main/README.md).
- [x] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/abetlen/llama-cpp-python/discussions), and have a new bug or useful enhancement to share.

# Observed Behavior

Setting `verbose = False` when creating a `Llama` instance causes all program output to disappear while the model is running. This includes in particular `logging` output from other threads. In addition, if I create several `Llama` instances and use them in parallel from multiple threads, I sometimes receive some (but not all) output from one of the instances (I assume that's because another instance finished running first and restored the output for everyone). And sometimes output is _not_ restored when all instances have finished (probably because the last instance to finish restored output to a blocked state).

# Expected Behaviour

Setting `verbose = False` when creating a `Llama` instance should cause output from that instance to disappear. Other program output should still be visible (this includes `logging` output and output from other `Llama` instances).

**Comment by jeberger** [comment]
As a workaround for `logging` on Linux, you can duplicate the `stderr` file descriptor and log to _that_ instead of the true `stderr`. E.g.:

```python
logging.basicConfig (
    level = LOG_LEVEL, format = LOG_FORMAT,
    handlers = [
        logging.StreamHandler (os.fdopen (
            os.dup (sys.stderr.fileno()), 'w'))
    ])
```

This allows logging to proceed even while output has been blocked by llama.cpp.
